src/httpserver.cpp

```diff
@@ -459,14 +477,22 @@ void StopHTTPServer()
         evhttp_del_accept_socket(eventHTTP, socket);
     }
     boundSockets.clear();
+    {
+        WAIT_LOCK(g_requests_mutex, lock);
+        while (!g_requests.empty()) {
+            g_requests_cv.wait(lock);
```

Comment:
maybe log "Waiting for %d requests to stop HTTP server" or something like that.

Reply:
done